ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -414,6 +421,19 @@ private SafeFuture<BlockImportResult> onBlock(
 
     blobSidecarsAvailabilityChecker.initiateDataAvailabilityCheck();
 
+    final SafeFuture<BlobSidecarsAndValidationResult> blobSidecarsAvailabilityResult =
```

Comment:
My main concern on this that we had block processing, which is not fast and only after that were checking the state of blobSidecar validation. So we had a chance it was run simultaneously (correct me if it's on one thread), now we are blocked by blobSidecar validation.

Reply:
Consensus validation IS a successful state transition. So I think the bug here is that the consensus validation future can be completed before the state transition is happening.